### ISSUE TITLE
deeper binding question

### ISSUE THREAD

**Issue by aktxyz** [rsamec/react-binding#1]
First off I really like your project.

It looks like the structure of the initial state does not need to be defined up front, for example in the hobby sample, the initial state is basically an empty data object.  As you type in the form fields, the structure of the data object is filled in, and you can see that in your pretty json.

``` javascript
  getInitialState: function() {
    return { data: {}};
  },
```

I tried to see how far that would go, and created a new plnkr that added 2 more inputs

``` javascript
        <TextBoxInput label="Street"  model={this.bindTo(this.props.personModel,"Address.Street")} />
        <TextBoxInput label="Phone"  model={this.bindTo(this.props.personModel,"Phone.Main.Number")} />
```

The 2 level deep object was filled in as expected, just like your existing inputs.  The 3 level deep object did not fill in.  This may be expected, but not sure.  It would be slick if that worked as well.

edit ...
Adding the plnkr url for the updated sample

http://plnkr.co/edit/802962ElwKUd8tPFYan2?p=preview

**Comment by rsamec** [comment]
Yes, you are right. This is not expected for the 3 level deep object. I will have a look at this issue at the weekend.

You can use workaround below - to define empty object up front.

``` js
getInitialState: function() {
    return { data: {Person:{Phone:{}}}};
  },
```

**Comment by rsamec** [comment]
Fixed in react-binding@0.6.0.
